index.html

```diff
@@ -1213,59 +1216,94 @@
       <h2>
         Examples
       </h2>
-      <h3>
-        Playback of depth and color streams from same device group.
-      </h3>
-      <pre class="example">
-navigator.mediaDevices.getUserMedia({
-  video: {videoKind: {exact: "color"}, groupId: {exact: id}}
-}).then(function (stream) {
-    // Wire the media stream into a &lt;video&gt; element for playback.
-    // The RGB video is rendered.
-    var video = document.querySelector('#video');
-    video.srcObject = stream;
-    video.play();
+      <aside title="Playback of depth and color streams from same device group"
+      class="example">
+        <pre class="highlight">
+async function attachVideoStream(el, kind, groupId) {
+  const constraints = {
+    video: {
+      mandatory: {
+        groupId: groupId
+      },
+      videoKind: { exact: kind }
+    }
   }
-);
 
-navigator.mediaDevices.getUserMedia({
-  video: {videoKind: {exact: "depth"}, groupId: {exact: id}}
-}).then(function (stream) {
-    // Wire the depth-only stream into another &lt;video&gt; element for playback.
-    // The depth information is rendered in its grayscale representation.
-    var depthVideo = document.querySelector('#depthVideo');
-    depthVideo.srcObject = stream;
-    depthVideo.play();
-  }
-);
+  const stream = await navigator.mediaDevices.getUserMedia(constraints);
+
+  el.srcObject = stream;
+  el.play();
+
+  return el;
+}
+
+async function play() {
+  const colorEl = document.createElement("video");
+  const depthEl = document.createElement("video");
+
+  const body = document.querySelector('body');
+  body.appendChild(colorEl);
+  body.appendChild(depthEl);
+
+  // Assume that all our video inputs are depth stream capable.
```

Comment:
We can use videoKind to avoid the need for this assumptions - usually there is an integrated camera and this approach (via navigator.mediaDevices.enumerateDevices()) didn't work. I guess counting the MediaDeviceInfo that are videoinput with the same groupId would give a closer match but it still looks like a hack.
  
We shouldn't need to use enumerateDevices - first use videoKind to get depth stream,  then get groupId from the depthStream's track and then use videoKind: color + groupId to get corresponding color device.
I didn't try this yet - it depends on experimental MediaStreamTrack.getSettings() - https://crbug.com/681824. Maybe it is already available through MediaTrackCapabilities though.

The implementation for videoKind has recently been added to Chromium: https://crrev.com/2664673002/.